state/accountsDBApi.go

```diff
@@ -99,17 +133,25 @@ func (accountsDB *accountsDBApi) RevertToSnapshot(_ int) error {
 
 // GetCode will call the inner accountsAdapter method after trying to recreate the trie
 func (accountsDB *accountsDBApi) GetCode(codeHash []byte) []byte {
-	err := accountsDB.recreateTrieIfNecessary()
+	code, _, err := accountsDB.GetCodeWithBlockInfo(codeHash)
 	if err != nil {
-		return nil
+		log.Warn("accountsDBApi.GetCode", "error", err)
 	}
 
-	return accountsDB.innerAccountsAdapter.GetCode(codeHash)
+	return code
 }
 
 // RootHash will return last loaded root hash
 func (accountsDB *accountsDBApi) RootHash() ([]byte, error) {
-	return accountsDB.trieController.getLatestRootHash()
+	accountsDB.mutBlockInfo.RLock()
+	defer accountsDB.mutBlockInfo.RUnlock()
+
+	blockInfo := accountsDB.blockInfo
```

Comment:
I think we could have only wrapped the access to this member in the critical section. Fine as it is, though.